### PR TITLE
Escape benchmark comment title before matching

### PR DESCRIPTION
**Pull request by pzread**
Escape the title before matching with the previous comment posts, so the `_` in the title won't cause the matching failed.

### CODE REVIEW

**Review by GMNGeoffrey — APPROVED** [review]

File: build_tools/buildkite/cmake/linux/x86_64/benchmark.yml [hunk 1]
```diff
@@ -46,7 +46,7 @@ steps:
     commands:
       - "git clean -fdx"
       - "buildkite-agent artifact download benchmark-results-*.json ./"
-      - "python3 build_tools/benchmarks/post_benchmarks_as_pr_comment.py --verbose --query-base --comment-title=\"Abbreviated x86_64 Benchmark Summary (experimental)\" benchmark-results-*.json"
+      - "python3 build_tools/benchmarks/post_benchmarks_as_pr_comment.py --verbose --query-base --comment-title=\"Abbreviated x86-64 Benchmark Summary (experimental)\" benchmark-results-*.json"
```
Comment (ScottTodd) [inline build_tools/buildkite/cmake/linux/x86_64/benchmark.yml:49]: Even though this is just the comment title, I'd still rather stick to `x86_64` matching the rest of the project.

Is the issue that if you escape the underscore (which in markdown indicates the start of italics), this search will fail?

https://github.com/google/iree/blob/2a646dbf6107c8690de5e0a63bfde0d9bfdb6a6f/build_tools/benchmarks/post_benchmarks_as_pr_comment.py#L246-L250

Comment (pzread) [reply build_tools/buildkite/cmake/linux/x86_64/benchmark.yml:49]: Yes, that is the issue. The posted comment has `x86\_64` in the title, so this match fails.

Comment (pzread) [reply build_tools/buildkite/cmake/linux/x86_64/benchmark.yml:49]: I guess I can extract the title from the generated markdown and use it to match, if we want to use `x86_64` in the title.

Comment (ScottTodd) [reply build_tools/buildkite/cmake/linux/x86_64/benchmark.yml:49]: Or try using `esc_format()`: https://github.com/awesmubarak/markdown_strings#helper-functions

**Review by ScottTodd — APPROVED** [review]
Looks like that worked?

**Comment by pzread** [comment]
> Looks like that worked?

Yes, thanks for the suggestions : )

**Pull request merged**